src/cross_section.F90

```diff
@@ -133,6 +142,15 @@ subroutine calculate_xs(p)
       ! Add contributions to material macroscopic energy release from fission
       material_xs % kappa_fission = material_xs % kappa_fission + &
            atom_density * micro_xs(i_nuclide) % kappa_fission
+
+      ! Add contributions to material macroscopic delayed-nu-fission cross
+      ! section
+      do d = 1, nuclides(i_nuclide) % n_precursor
+        yield = yield_delayed(nuc, p % E, d)
+        material_xs % delayed_nu_fission(d) = &
+             material_xs % delayed_nu_fission(d) +&
+             atom_density * micro_xs(i_nuclide) % delayed_nu_fission * yield
+      end do
```

Comment:
Unfortunately this block of code here is a showstopper. MC code performance is particularly sensitive to cross section lookups, and this code makes that lookup quite a bit more onerous. I just ran a simple test on the Hoogenboom-Martin benchmark and the performance drops by more than 50%.

My suggestion is to calculate these micro and macroscopic cross sections on-the-fly when needed rather than calculating them up-front here every time the particle's energy changes and storing them in the micro/macro caches. Otherwise, this hurts performance pretty severely for a simulation that doesn't even use your new score/filter.